ext/mbstring/libmbfl/mbfl/mbfilter.c

```diff
@@ -865,6 +871,23 @@ mbfl_strpos(
 		goto out;
 	}
 
+	if (needle_u8->len == 0) {
+		if (reverse) {
+			if (offset < 0) {
+				result = (size_t) -offset;
```

Comment:
This case doesn't look right to me. Actually, why is `reverse` a separate case at all? Doesn't the offset refer to the same position regardless of reverse or not?

Reply:
Good question ... Maybe was just overcomplicating it on my own. Will have a look when I get back

Reply:
So reverse is a special case because if offset is 0 instead of getting the last code point you get the first one, now as to this case specifically this seems indeed to be a mistake and should probably be more ``haystack_length + offset`` to get the codepoint ``offset`` codepoints from the end. because currently it gives the offset codepoint like the strpos case

Reply:
Oooh, yes, that makes sense. I think the correct values are `haystack_length + offset` for negative offset and just `haystack_length` for positive offset.

Reply:
Ah yeah indeed, will fix that and commit.